File: 2trabajoConstructores/js/clase normal/AnimalSinParametros.js
class AnimalSinParametros {
    constructor() {
        this.nombre = "Desconocido";
        this.sonido = "Sonido indefinido";
    }

    hacerSonido() {
        console.log(`${this.nombre} hace: ${this.sonido}`);
    }
}

module.exports = AnimalSinParametros;

File: 2trabajoConstructores/go/interfaz/animal_sin_parametros.go
package main

import "fmt"

type AnimalSinParametros struct {
	Nombre string
	Sonido string
}

// "Constructor"
func NuevoAnimalSinParametros() AnimalSinParametros {
	return AnimalSinParametros{
		Nombre: "Desconocido",
		Sonido: "Ninguno",
	}
}

// Implementa la interfaz IAnimal
func (a AnimalSinParametros) HacerSonido() {
	fmt.Printf("%s hace: %s\n", a.Nombre, a.Sonido)
}
